### PR TITLE
ci(helm): create helm integration test workflow

### PR DESCRIPTION
**Pull request by DenizParlak**
Because

- We need an automated helm integration test.

This commit

- Contains GA workflow to create minikube cluster & test it.

### CODE REVIEW

File: scr [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+export APIGATEWAY_POD_NAME=$(kubectl get pods --namespace vdp -l "app.kubernetes.io/component=api-gateway,app.kubernetes.io/instance=vdp" -o jsonpath="{.items[0].metadata.name}")
+
+export APIGATEWAY_CONTAINER_PORT=$(kubectl get pod --namespace vdp $APIGATEWAY_POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
+
+export CONSOLE_POD_NAME=$(kubectl get pods --namespace vdp -l "app.kubernetes.io/component=console,app.kubernetes.io/instance=vdp" -o jsonpath="{.items[0].metadata.name}")
+
+export CONSOLE_CONTAINER_PORT=$(kubectl get pod --namespace vdp $CONSOLE_POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
+
+sleep 7
```
Comment (pinglin) [inline scr:24]:
```suggestion
```

File: scr [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+#helm repo add instill https://helm.instill.tech
+#helm repo update
+#helm install vdp instill/vdp --devel --namespace $namespace --create-namespace &>/dev/null
+
+
+helm install vdp . --devel --namespace $namespace --create-namespace --set enableITMode=true  &>/dev/null 
+
+
+sleep 15
```
Comment (pinglin) [inline scr:14]:
```suggestion
export MODEL_POD_NAME=$(kubectl get pods --namespace vdp -l "app.kubernetes.io/component=model,app.kubernetes.io/instance=vdp" -o jsonpath="{.items[0].metadata.name}")
while [[ $(kubectl get pods $MODEL_POD_NAME -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}' --namespace vdp) != "True" ]]; do echo "waiting for pod" && sleep 1; done
```

File: scr [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+# Instill/VDP Helm chart'ının yükleneceği Kubernetes namespace'ini belirleyin.
+namespace="vdp"
+
+#helm repo add instill https://helm.instill.tech
+#helm repo update
+#helm install vdp instill/vdp --devel --namespace $namespace --create-namespace &>/dev/null
+
+
+helm install vdp . --devel --namespace $namespace --create-namespace --set enableITMode=true  &>/dev/null
```
Comment (pinglin) [inline scr:11]: At this point the script path is in the repo root? Should move to `charts/vdp` to make it work.

File: scr [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+
+pod_gw=$(kubectl get po -n vdp --no-headers=true | awk '{print $1}' | grep gateway)
+pod_console=$(kubectl get po -n vdp --no-headers=true | awk '{print $1}' | grep console)
+
+curl --silent --fail http://localhost:3000 &>/dev/null
+echo "Console OK"
+
+curl --silent --fail http://localhost:8080 &>/dev/null
+echo "API OK"
```
Comment (pinglin) [inline scr:71]:
```suggestion
echo "api-gateway OK"
```

File: scr [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+echo " "
+
+echo "All remaining pods are running."
+
+
+pod_gw=$(kubectl get po -n vdp --no-headers=true | awk '{print $1}' | grep gateway)
+pod_console=$(kubectl get po -n vdp --no-headers=true | awk '{print $1}' | grep console)
+
+curl --silent --fail http://localhost:3000 &>/dev/null
+echo "Console OK"
```
Comment (pinglin) [inline scr:68]:
```suggestion
echo "console OK"
```

**Review by pinglin — CHANGES_REQUESTED** [review]
The pod of `model-backend` should be always the very last one to be `RUNNING`. We'd better just wait for its health signal instead of using `sleep`.

File: scr (hunk 3 above)
Comment (DenizParlak) [reply scr]: yes, I used helm repo directly sometimes to compare it with local helm environment.

**Pull request closed without merging**